group_project_v2/project_api/api_implementation.py

```diff
@@ -399,6 +392,25 @@ def get_member_data(self, user_id):
         return user_details
 
     @memoize_with_expiration()
+    def get_user_roles_for_course(self, user_id, course_id):
+        """
+        Returns role names user has for a given course.
+
+        :param int user_id: User Id
+        :param int course_id: Course id
+
+        :rtype: set[str]
+        """
+
+        qs_params = {
+            "user_id": user_id,
+        }
+        return set(
+            role['role']
+            for role in self.send_request(GET, (COURSES_API, course_id, 'roles'), query_params=qs_params)
```

Comment:
@jbzdak nit: it looks like it would look slightly better as follows:

    response = self.send_request(GET, (COURSES_API, course_id, 'roles'), query_params=qs_params)
    return  set(role['role'] for role in response)